cdap-ui/app/directives/dag-plus/my-dag-factory.js

```diff
@@ -56,39 +43,51 @@ angular.module(PKG.name + '.commons')
       },
       anchors: [ 'Static']
     };
-    var sourceSettings = angular.extend({
+    var originSettings = angular.extend({
       isSource: true,
       connectorStyle: connectorStyle,
-      anchor: [ 0.5, 1, 1, 0, 26, -43, 'sourceAnchor']
+      connectorOverlays: [
+        [ 'Arrow', { location: 1, length: 12, width: 12, height: 10, foldback: 1 } ],
+      ],
+      anchor: [ 0.5, 1, 1, 0, 26, -43 ]
     }, commonSettings);
-    var sinkSettings = angular.extend({
+    var targetSettings = angular.extend({
       isTarget: true,
-      anchor: [ 0.5, 1, -1, 0, -26, -43, 'sinkAnchor'],
+      connectorOverlays: [
+        [ 'Arrow', { location: 1, length: 12, width: 12, height: 10, foldback: 1 } ],
+      ],
+      anchor: [ 0.5, 1, -1, 0, -27, -43 ],
       connectorStyle: connectorStyle
     }, commonSettings);
 
-    function getSettings(isDisabled) {
-      var settings = {};
-      if (isDisabled) {
-        settings = {
-          default: defaultSettings,
-          commonSettings: angular.extend(commonSettings, disabledConnectorOverlays),
-          source: angular.extend(sourceSettings, disabledConnectorOverlays),
-          sink: angular.extend(sinkSettings, disabledConnectorOverlays)
-        };
-      } else {
-        settings = {
-          default: defaultSettings,
-          commonSettings: angular.extend(commonSettings, connectorOverlays),
-          source: angular.extend(sourceSettings, connectorOverlays),
-          sink: angular.extend(sinkSettings, connectorOverlays)
-        };
-      }
-
-      settings.transformSource = angular.copy(settings.source);
-      settings.transformSink = angular.copy(settings.sink);
-      settings.transformSource.anchor = [ 0.5, 1, 1, 0, 26, -43, 'transformAnchor'];
-      settings.transformSink.anchor = [ 0.5, 1, -1, 0, -26, -43, 'transformAnchor'];
+    function getSettings() {
+      var settings = {
+        default: defaultSettings,
+        commonSettings: angular.extend(commonSettings, connectorOverlays),
+        sourceOrigin: angular.copy(originSettings),
```

Comment:
If we are doing `angular.copy` while performing `getSettings` do we need to do it here? We should try doing it in one place once..